File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    clothesDict = {}
    
    for name, classs in clothes:
        clothesDict[classs] = clothesDict.get(classs, 0) + 1
        
    ans = 1
    
    for classs in clothesDict:
        ans *= clothesDict[classs] + 1
        
    return ans - 1
    
    

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(int i=0; i<clothes.length; i++){
            hm.put(clothes[i][1], hm.getOrDefault(clothes[i][1], 0) + 1);  
        }
        
        int answer = 1;
        
        for(String key : hm.keySet()){
            answer *= hm.get(key) + 1;
            //System.out.println(hm.get(key));
        }
        return answer - 1;
    }
}